node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -75,12 +75,12 @@ impl ScrapedUpdates {
 /// in each vector are ordered by decreasing parent block number to facilitate
 /// minimal cost pruning.
 pub struct Inclusions {
-	inclusions_inner: HashMap<CandidateHash, Vec<(BlockNumber, Hash)>>,
+	inclusions_inner: BTreeMap<CandidateHash, BTreeMap<BlockNumber, Vec<Hash>>>,
 }
 
 impl Inclusions {
 	pub fn new() -> Self {
-		Self { inclusions_inner: HashMap::new() }
+		Self { inclusions_inner: BTreeMap::new() }
 	}
 
 	// Insert parent block into vector for the candidate hash it is including,
```

Comment:
`BTreeMap` maintains elements in ascending order (doc says "decreasing", 1 line below here).

Reply:
Thanks for updating the doc! I'm still a little confused why we store hashes in a vec - can there be more than one block hash at a block number? Again, I'm probably just being slow today.

Reply:
To my understanding we can have multiple relay chain blocks at the same height on different forks which include the same parachain block candidate. So the vec contains all of them. Though in most cases I'd expect each vec to have only one entry.